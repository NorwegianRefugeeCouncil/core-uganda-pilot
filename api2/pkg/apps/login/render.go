package login

import "net/http"

func (s *Server) Render(w http.ResponseWriter, req *http.Request, templateName string, data map[string]interface{}) {
	if err := s.template.ExecuteTemplate(w, templateName, data); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
}
