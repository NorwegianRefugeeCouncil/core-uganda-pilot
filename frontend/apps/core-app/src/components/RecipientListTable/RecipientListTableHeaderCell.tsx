import { HStack, IconButton, Text } from 'native-base';
import { Icon } from 'core-design-system';
import React from 'react';
import { ColumnInstance, UseSortByColumnProps } from 'react-table';

import { RecipientListTableEntry } from './types';

type Props<T extends Record<string, any>> = {
  column: ColumnInstance & UseSortByColumnProps<T>;
};

export const RecipientListTableHeaderCell: React.FC<
  Props<RecipientListTableEntry>
> = ({ column }) => {
  const sortIcon = column.isSorted
    ? column.isSortedDesc
      ? 'more'
      : 'arrowUp'
    : 'arrowDown';

  const handleSortToggle = () => column.toggleSortBy(!column.isSortedDesc);

  return (
    <HStack
      width={column.width}
      p="2"
      alignItems="center"
      justifyContent="space-between"
      flexGrow={1}
    >
      {column.render(Text, {
        fontWeight: 'bold',
        fontSize: 'xs',
        lineHeight: '4xs',
        children: column.Header,
      })}
      <IconButton
        onPress={handleSortToggle}
        colorScheme="secondary"
        variant="ghost"
        size="sm"
        icon={<Icon size={5} name={sortIcon} />}
      />
    </HStack>
  );
};
