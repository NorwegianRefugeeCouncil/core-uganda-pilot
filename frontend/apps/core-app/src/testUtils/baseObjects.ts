import { FormDefinition, FormType, Record } from 'core-api-client';

export const baseForm: FormDefinition = {
  id: 'form1',
  code: '',
  name: 'form 1',
  databaseId: 'databaseId1',
  folderId: 'folderId1',
  formType: FormType.DefaultFormType,
  fields: [],
};

export const baseTextField = {
  id: 'field1',
  name: 'field 1',
  description: 'description 1',
  code: '',
  required: false,
  key: false,
  fieldType: {
    text: {},
  },
};

export const baseMultilineTextField = {
  id: 'field2',
  name: 'field 2',
  description: 'description 2',
  code: '',
  required: true,
  key: true,
  fieldType: {
    multilineText: {},
  },
};

export const baseRecord1: Record = {
  id: 'record1',
  values: [
    {
      value: 'text',
      fieldId: 'field1',
    },
    {
      value: 'multi line text',
      fieldId: 'field2',
    },
  ],
  formId: 'form1',
  databaseId: 'databaseId1',
  ownerId: undefined,
};
