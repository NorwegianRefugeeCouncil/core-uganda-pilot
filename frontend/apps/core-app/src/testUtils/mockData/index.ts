export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
