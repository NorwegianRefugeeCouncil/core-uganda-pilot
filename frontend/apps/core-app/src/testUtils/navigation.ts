export const mockNavigationProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
};
