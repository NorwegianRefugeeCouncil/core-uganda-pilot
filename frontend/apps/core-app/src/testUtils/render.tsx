import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
