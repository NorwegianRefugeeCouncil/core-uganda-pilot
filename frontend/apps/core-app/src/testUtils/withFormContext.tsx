import { Button } from 'native-base';
import React from 'react';
import { FieldValues, FormProvider, useForm } from 'react-hook-form';

export const withFormContext =
  <T,>(
    Component: React.FC<T>,
    defaultValues: FieldValues = {},
    onSubmit: (data: any) => void = () => {},
  ) =>
  // eslint-disable-next-line react/display-name
  (props: T) => {
    const f = useForm({ defaultValues });

    const handleSubmit = f.handleSubmit((data: any) => {
      onSubmit(data);
    });

    return (
      <FormProvider {...f}>
        <Component {...props} />
        <Button testID="with-form-context-submit-button" onPress={handleSubmit}>
          Submit
        </Button>
      </FormProvider>
    );
  };
