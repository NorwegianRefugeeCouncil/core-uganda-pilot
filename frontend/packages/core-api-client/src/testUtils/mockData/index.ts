export * from './Form';
export * from './FormField';
export * from './Record';
