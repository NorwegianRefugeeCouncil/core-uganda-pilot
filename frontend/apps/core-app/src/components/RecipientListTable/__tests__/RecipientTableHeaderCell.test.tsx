import React from 'react';
import { act, fireEvent, waitFor } from '@testing-library/react-native';

import { render } from '../../../testUtils/render';
import { RecipientListTableHeaderCell } from '../RecipientListTableHeaderCell';

describe('RecipientTableHeaderCell', () => {
  describe('should match the snapshot', () => {
    it('unsorted', () => {
      const { toJSON } = render(
        <RecipientListTableHeaderCell
          column={{
            getSortByToggleProps: () => ({ onClick: jest.fn() }),
            isSorted: false,
            isSortedDesc: false,
            render: jest.fn(),
            toggleSortBy: jest.fn(),
          }}
        />,
      );
      expect(toJSON()).toMatchSnapshot();
    });
    it('sorted, descending', () => {
      const { toJSON } = render(
        <RecipientListTableHeaderCell
          column={{
            getSortByToggleProps: () => ({ onClick: jest.fn() }),
            isSorted: true,
            isSortedDesc: true,
            render: jest.fn(),
            toggleSortBy: jest.fn(),
          }}
        />,
      );
      expect(toJSON()).toMatchSnapshot();
    });
    it('sorted, ascending', () => {
      const { toJSON } = render(
        <RecipientListTableHeaderCell
          column={{
            getSortByToggleProps: () => ({ onClick: jest.fn() }),
            isSorted: true,
            isSortedDesc: false,
            render: jest.fn(),
            toggleSortBy: jest.fn(),
          }}
        />,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should call press handler', () => {
    const toggleHandler = jest.fn();
    const { getByTestId } = render(
      <RecipientListTableHeaderCell
        column={{
          getSortByToggleProps: () => ({ onClick: jest.fn() }),
          isSorted: true,
          isSortedDesc: false,
          render: jest.fn(),
          toggleSortBy: toggleHandler,
        }}
      />,
    );
    const button = getByTestId('recipient-table-sort-button');
    act(() => {
      fireEvent.press(button);
    });
    waitFor(() => {
      expect(toggleHandler).toHaveBeenCalled();
    });
  });
});
