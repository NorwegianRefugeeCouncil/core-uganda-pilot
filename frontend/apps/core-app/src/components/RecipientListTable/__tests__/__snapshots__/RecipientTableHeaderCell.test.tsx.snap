// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipientTableHeaderCell should match the snapshot sorted, ascending 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "space-between",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <View
      _text={
        Object {
          "color": "lightText",
        }
      }
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="secondary"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onHoverIn={[Function]}
      onHoverOut={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 4,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
      testID="recipient-table-sort-button"
    >
      <RNSVGSvgView
        accessibilityRole="image"
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        color={4280690214}
        dataSet={Object {}}
        fill="currentColor"
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke=""
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "color": "#262626",
              "height": 20,
              "width": 20,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        testID="sort-icon-arrowUp"
        tintColor={4280690214}
        vbHeight={20}
        vbWidth={20}
      >
        <RNSVGGroup
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
              "stroke",
            ]
          }
          stroke={null}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M5.41 13.7051L10 9.12508L14.59 13.7051L16 12.2951L10 6.29508L4 12.2951L5.41 13.7051Z"
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`RecipientTableHeaderCell should match the snapshot sorted, descending 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "space-between",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <View
      _text={
        Object {
          "color": "lightText",
        }
      }
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="secondary"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onHoverIn={[Function]}
      onHoverOut={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 4,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
      testID="recipient-table-sort-button"
    >
      <RNSVGSvgView
        accessibilityRole="image"
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        color={4280690214}
        dataSet={Object {}}
        fill="currentColor"
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke=""
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "color": "#262626",
              "height": 20,
              "width": 20,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        testID="sort-icon-more"
        tintColor={4280690214}
        vbHeight={20}
        vbWidth={20}
      >
        <RNSVGGroup
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
              "stroke",
            ]
          }
          stroke={null}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M9.99981 5C11.3599 5 12.4728 3.875 12.4728 2.5C12.4728 1.125 11.3599 0 9.99981 0C8.63968 0 7.52686 1.125 7.52686 2.5C7.52686 3.875 8.63968 5 9.99981 5ZM9.99981 7.49976C8.63968 7.49976 7.52686 8.62476 7.52686 9.99976C7.52686 11.3748 8.63968 12.4998 9.99981 12.4998C11.3599 12.4998 12.4728 11.3748 12.4728 9.99976C12.4728 8.62476 11.3599 7.49976 9.99981 7.49976ZM9.99981 15.0001C8.63968 15.0001 7.52686 16.1251 7.52686 17.5001C7.52686 18.8751 8.63968 20.0001 9.99981 20.0001C11.3599 20.0001 12.4728 18.8751 12.4728 17.5001C12.4728 16.1251 11.3599 15.0001 9.99981 15.0001Z"
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`RecipientTableHeaderCell should match the snapshot unsorted 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "space-between",
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      }
    }
  >
    <View
      _text={
        Object {
          "color": "lightText",
        }
      }
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="secondary"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onHoverIn={[Function]}
      onHoverOut={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 4,
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        }
      }
      testID="recipient-table-sort-button"
    >
      <RNSVGSvgView
        accessibilityRole="image"
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        color={4280690214}
        dataSet={Object {}}
        fill="currentColor"
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke=""
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "color": "#262626",
              "height": 20,
              "width": 20,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        testID="sort-icon-arrowDown"
        tintColor={4280690214}
        vbHeight={20}
        vbWidth={20}
      >
        <RNSVGGroup
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
              "stroke",
            ]
          }
          stroke={null}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M14.59 6.29492L10 10.8749L5.41 6.29492L4 7.70492L10 13.7049L16 7.70492L14.59 6.29492Z"
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
